### PR TITLE
feat: Website unavailable error page

### PR DESCRIPTION
**Pull request by Yserz**
image

Closes #3564

### CODE REVIEW

File: electron/renderer/src/components/App.jsx [hunk 1]
```diff
@@ -49,9 +55,16 @@ class App extends React.Component {
     const accountId = this.props.accountIds[accountIndex];
     if (accountId) {
       this.props.switchAccount(accountId);
-    }
 
-    event.target.focus();
+      // Note: We need to focus window first to properly set focus
+      // on the webview with shortcuts like Cmd+1/2/3
+      window.blur();
+      window.focus();
```
Comment (lipis) [inline electron/renderer/src/components/App.jsx]: the magic lines!

File: electron/renderer/src/components/IsOnline.jsx [hunk 2]
```diff
@@ -17,26 +17,31 @@
...
-      window.addEventListener('online', () => this.setState({isOnline: true}), {once: true});
-    }
-  }
+  return isOnline ? (
+    children
+  ) : (
+    <div style={{display: 'flex', height: '100%', width: '100%'}}>
+      <ContainerSM centerText verticalCenter>
+        <Text center block textTransform="uppercase">
+          {'No Internet'}
```
Comment (lipis) [inline electron/renderer/src/components/IsOnline.jsx]: No translations here?

File: electron/renderer/src/components/App.jsx [hunk 3]
```diff
@@ -37,6 +38,11 @@ class App extends React.Component {
   componentDidMount() {
     window.addEventListener(EVENT_TYPE.ACTION.SWITCH_ACCOUNT, this.switchAccount, false);
     window.addEventListener(EVENT_TYPE.ACTION.CREATE_SSO_ACCOUNT, this.initiateSSO, false);
+
+    // Workaround: Switch to first webview after startup
```
Comment (Yserz) [inline electron/renderer/src/components/App.jsx]: That's for focusing the correct webview and placing the cursor

File: electron/renderer/src/components/IsOnline.jsx (hunk 2 above)
Comment (Yserz) [reply electron/renderer/src/components/IsOnline.jsx]: There was no translation logic for the renderer process before. There should be a separate PR for adding translations from the renderer process.

File: electron/renderer/src/components/App.jsx (hunk 3 above)
Comment (Yserz) [reply electron/renderer/src/components/App.jsx]: Improvement https://github.com/wireapp/wire-desktop/pull/3569

File: electron/renderer/src/components/App.jsx [hunk 4]
```diff
@@ -37,6 +38,11 @@ class App extends React.Component {
...
+    setTimeout(() => {
+      this.switchAccount({detail: {accountIndex: 0}});
+    }, 1000);
```
Comment (bennyn) [inline electron/renderer/src/components/App.jsx]: Do we have other chances than setting a timeout?

File: electron/renderer/src/components/Webview.jsx [hunk 5]
```diff
@@ -19,60 +19,214 @@
...
-    if (nextProps.visible) {
-      this._focusWebview();
+      return url.href;
+    };
+    const newUrl = getEnvironmentUrl();
+    if (url !== newUrl && webviewRef.current) {
+      setUrl(newUrl);
+      try {
+        // can't load URL before webview is not attached to the DOM
+        webviewRef.current.loadURL(newUrl).catch(error => console.info(`Navigating to ${newUrl} failed`, error));
```
Comment (bennyn) [inline electron/renderer/src/components/Webview.jsx]:
```suggestion
        webviewRef.current.loadURL(newUrl).catch(error => console.error(`Navigating to ${newUrl} failed`, error));
```

File: electron/renderer/src/components/Webview.jsx [hunk 6]
```diff
@@ -19,60 +19,214 @@
...
+      } catch (error) {
+        console.info('Can not #loadURL before attaching webview to DOM', error);
```
Comment (bennyn) [inline electron/renderer/src/components/Webview.jsx]:
```suggestion
        console.error(`Can not ${newUrl} before attaching webview to DOM`, error);
```

File: electron/src/renderer/preload-app.ts [hunk 7]
```diff
@@ -127,10 +127,10 @@ setupIpcInterface();
 subscribeToMainProcessEvents();
 
 window.addEventListener('DOMContentLoaded', addDragRegion);
-
 window.addEventListener('focus', () => {
   const selectedWebview = getSelectedWebview();
   if (selectedWebview) {
+    selectedWebview.blur();
```
Comment (bennyn) [inline electron/src/renderer/preload-app.ts]: Is it the same trick used in `App.jsx`? If yes, let's put it in a utility to remember and reuse it.

File: electron/src/renderer/preload-webview.ts [hunk 8]
```diff
@@ -212,21 +194,26 @@ process.once('loaded', () => {
...
+    logger.info('Registering event handlers');
     subscribeToMainProcessEvents();
     subscribeToThemeChange();
     subscribeToWebappEvents();
     reportWebappVersion();
-    // include context menu
-    import('./menu/context').catch(error => logger.error(error));
   });
+  // include context menu
+  import('./menu/context').catch(error => logger.error(error));
```
Comment (bennyn) [inline electron/src/renderer/preload-webview.ts]:
```suggestion
  import('./menu/context').catch(logger.error);
```

File: electron/renderer/src/components/App.jsx (hunk 4 above)
Comment (Yserz) [reply electron/renderer/src/components/App.jsx]: I haven't found another solution yet

File: electron/renderer/src/components/Webview.jsx (hunk 6 above)
Comment (Yserz) [inline electron/renderer/src/components/Webview.jsx]:
```suggestion
        console.warn('Can not #loadURL before attaching webview to DOM', error);
```

File: electron/src/renderer/preload-app.ts (hunk 7 above)
Comment (Yserz) [reply electron/src/renderer/preload-app.ts]: No, that one is different

**Pull request merged**